lib/EarlyFunctionAnalysis/DetectABIMFI.h

```diff
@@ -0,0 +1,212 @@
+#pragma once
+
+#include <map>
+
+#include "llvm/IR/GlobalVariable.h"
+
+#include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
+#include "revng/EarlyFunctionAnalysis/CallGraph.h"
+#include "revng/EarlyFunctionAnalysis/Outliner.h"
+#include "revng/EarlyFunctionAnalysis/CFGAnalyzer.h"
+#include "revng/MFP/MFP.h"
+#include "revng/MFP/SetLattices.h"
+#include "revng/Model/Register.h"
+
+#include "ABIAnalyses/Analyses.h"
+
+namespace efa {
+
+struct FunctionABI {
+  using SetOfRegisters = std::set<model::Register::Values>;
+  using SUL = SetUnionLattice<SetOfRegisters>;
+
+  SetOfRegisters ArgumentRegisters;
+  SetOfRegisters ReturnRegisters;
+
+  [[nodiscard]] FunctionABI combineValues(const FunctionABI &A2) const {
+    FunctionABI Result = {
+      .ArgumentRegisters = SUL::combineValues(ArgumentRegisters,
+                                              A2.ArgumentRegisters),
+      .ReturnRegisters = SUL::combineValues(ReturnRegisters, A2.ReturnRegisters)
+    };
+
+    return Result;
+  }
+
+  [[nodiscard]] bool isLessOrEqual(const FunctionABI &A2) const {
+    if (!SUL::isLessOrEqual(ArgumentRegisters, A2.ArgumentRegisters)) {
+      return false;
+    }
+
+    return SUL::isLessOrEqual(ReturnRegisters, A2.ReturnRegisters);
+  }
+};
+
+class LatticeElement {
```

Comment:
How is this different from `using LatticeElement = std::map<MetaAddress, FunctionABI>`?